support-services/src/main/scala/com/gu/support/redemption/DynamoTableAsync.scala

```diff
@@ -0,0 +1,86 @@
+package com.gu.support.redemption
+
+import java.util.concurrent.CompletionException
+
+import com.typesafe.scalalogging.LazyLogging
+import software.amazon.awssdk.auth.credentials.{AwsCredentialsProviderChain, InstanceProfileCredentialsProvider, ProfileCredentialsProvider}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, UpdateItemRequest}
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+object DynamoTableAsync {
+
+  val ProfileName = "membership"
+
+  lazy val CredentialsProvider =  AwsCredentialsProviderChain.builder.credentialsProviders(
+     ProfileCredentialsProvider.builder.profileName(ProfileName).build,
+     InstanceProfileCredentialsProvider.builder.asyncCredentialUpdateEnabled(false).build
+  ).build
+
+  def apply(table: String, key: String)(implicit e: ExecutionContext): DynamoTableAsync = {
+    val dynamoDBClient = DynamoDbAsyncClient.builder
+      .credentialsProvider(CredentialsProvider)
+      .region(Region.EU_WEST_1)
+      .build
+    new DynamoTableAsync(dynamoDBClient, table, key)
+  }
+}
+trait DynamoLookup {
+  def lookup(value: String): Future[Option[Map[String, Boolean]]]
```

Comment:
the implementation will discard all non-boolean attrributes. We can always extend it in future, but it didn't seem worth a full scala model, and the aws model is so verbose I wanted to keep it out of the API.
Could rename this to DynamoLookupBoolean in the meantime though?
I don't want to move too much code that should be unit tested into the client though.  As I don't plan to unit test the AWS api if at all possible!

Reply:
We are going to have to be able to return a `corporateAccountId` to store in Zuora as this is how we will link an account to a particular corporate customer.

My mocked implementation of this which I used to build the redemption form is here: https://github.com/guardian/support-frontend/pull/2518/files#diff-4d6b86fc69b601966db937093ee55092R105

It returns a case class with the redemption code and the account id (it actually has a corporate name as well but we are not actually using this currently)
https://github.com/guardian/support-frontend/pull/2518/files#diff-1f241c7b97cb23a815da6c5e5221eceeR5

Reply:
ok I will update suitably and make it support a string too

Reply:
so as far as I can tell for the client side we don't need any information other than is it valid, on the support-workers side we will need to know the `corporateAccountId` and whether it's valid.  If I'm reading the code and your comment correctly.
I am updating to return those from the underlying call, and use the appropriately.
If we do need extra information I'm happy to add, but starting with the simplest possible implementation, even serialising to json seems unnecessary on first sight?

Reply:
We will need the corporateAccountId in support-frontend but I think this interface should be fine.